Source_Code/Common/Messages/ActionRequests/SuggestRequest.java

```diff
@@ -0,0 +1,73 @@
+package Common.Messages.ActionRequests;
+
+import Common.CharacterCard;
+import Common.RoomCard;
+import Common.WeaponCard;
+import Common.PlayerHand;
+import Common.SolutionHand;
+import Common.Messages.ActionRequests.ActionRequest;
+
+public class SuggestRequest extends ActionRequest
+{
+    public String PlayerName;
+    public SolutionHand Hand;
+
+    public SuggestRequest()
+    {
+        super();
+    }
+
+    public SuggestRequest(String name, SolutionHand h)
+    {
+        this.PlayerName = name;
+        this.Hand = h;
+    }
+
+    public String toString()
+    {
+        String outString = "";
+        outString += this.PlayerName;
+        outString += "wants to suggest that ";
+        outString += this.Hand.getCharacters().get(0).getCharacterName();
+        outString += " dit it in the ";
+        outString += this.Hand.getRooms().get(0).getRoomName();
+        outString += " with the ";
+        outString += this.Hand.getWeapons().get(0).getWeaponType();
+        return outString;
+    }
+
+    // operation to check if a given player hand has any cards in it that will
+    // refute this suggestion request. If it does, it outputs those cards (as a
+    // new hand.) If not, it returns an empty hand.
+    public PlayerHand checkRefutations( PlayerHand playerHand )
+    {
+        PlayerHand outHand = new PlayerHand();
+
+        //probably a better way to do this than to loop through
+        for ( CharacterCard playerCard : playerHand.getCharacters())
+        {
+                if (playerCard.getCharacterName() == this.Hand.getCharacters().get(0).getCharacterName())
```

Comment:
getXName functions currently return strings, so these comparisons should be ...getXName().equals(...getXName()) instead of ==

Although I think those functions should probably be changed to return the enum values instead of strings

Reply:
left the legacy "name" functions because Im not sure what removing will break. (I know wrong answer)
added getXEnum and use that now.